script.js

```diff
@@ -0,0 +1,908 @@
+// Global Variables
+
+// ------- General -----//
+
+const playerHand = [];
+const commonCardsArray = [];
+// Tracks the number of same cards through each iteration of comparison
+const numCommonCardsArray = [];
+let shuffledDeck;
+
+// Track if any winning combi is true;
+let isAnyKindOrPair = false;
+let isFullHouse = false;
+let isFlush = false;
+let isStraight = false;
+let isStraightFlush = false;
+
+//  Global var that holds the name of the winning combi
+let nameOfWinCombi = 'Deal to begin.';
+
+// -------- Html Elements---------------------//
+let overallScreen;
+let payOutLevelPointer;
+let payOutScheduleDisplay;
+let cardsContainer;
+let statsDisplay;
+let buttonsContainer;
+
+// -------- Game Stats - Credits Mgmt & Display-----------//
+// Html elements for credits displays
+let creditsLeftDisplay;
+let outputMsgDisplay;
+let creditsInsertedDisplay;
+
+// Track number of credits
+let numCreditsInserted = 1;
+let creditsLeft = 100;
+let currAmtWon = 0;
+// Track payout level separate from creditsInserted
+let payOutLevel = 0;
+
+// Hardcoded individual credit payouts for X amount of credits used
+const oneCreditPayOut = [250, 50, 25, 9, 6, 4, 3, 2, 1, 0];
+const twoCreditPayout = [500, 100, 50, 18, 12, 8, 6, 4, 2, 0];
+const threeCreditPayout = [750, 150, 75, 27, 18, 12, 9, 6, 3, 0];
+const fourCreditPayout = [1000, 200, 100, 36, 24, 16, 12, 8, 4, 0];
+const fiveCreditPayout = [4000, 250, 125, 45, 30, 20, 15, 10, 5, 0];
+
+const payOutSchedule = [[...oneCreditPayOut], [...twoCreditPayout],
+  [...threeCreditPayout], [...fourCreditPayout], [...fiveCreditPayout]];
+
+// ----------------- Animation Management --------------//
+// Tracks new card images when swapped
+let newCardImage;
+// Tracks cardImages during new dealt
+let cardImage;
+// Tracks cover-card animations;
+let coverCardImage;
+let coverCardShown = true;
+
+// Track Rank of Hand
+// Five of a kind being 0 and Jacks or Better being 8, no winning hand = 9;
+let rankOfHand = 9;
+
+// --- Scroll Display Management ----//
+let combiDisplayCol;
+let pOut1Col;
```

Comment:
Could use some comments to explain the naming convention for these variables

Reply:
Ah I'm guessing `pOut` means `payOut`. Given that it's not a huge win to shorten var name by 2 chars, I personally might just use the full word `payOut` for clarity.

Reply:
Good point, sometimes perhaps I shld have commented it out for clarity? Because sometimes typing out 'payOutColumn' and 'combinationDisplayColumn' becomes quite verbose and I'm trying to reach a balance by shortening them.